src/coreclr/jit/emitxarch.cpp

```diff
@@ -5712,6 +5712,11 @@ void emitter::emitIns_R_I(instruction ins,
     id->idDebugOnlyInfo()->idMemCookie = targetHandle;
 #endif
 
+    if (emitComp->opts.IsReadyToRun() && EA_IS_CNS_SEC_RELOC(attr))
+    {
+        id->idAddr()->isSecRel = true;
+    }
+
```

Comment:
No, the handle are large and will not fit in small constant. I have added `assert(!id->idIsSmallDsc())` .

Reply:
You shouldn't depend on the constant values being "large enough".

In the case where you are setting `isSecRel`, you should instead create the ID using `emitNewInstrCns` instead of `emitNewInstrSC`.